### PR TITLE
feat(468): add artifact tree to build page

### PR DESCRIPTION
**Pull request by minz1027**
Add artifact tree to build page.
- There is a component `artifact-tree`
- Inside `artifact-tree`, it will call service `build-artifact` to fetch manifest data to build the tree
- Use [ember-cli-jstree](https://github.com/ritesh83/ember-cli-jstree) to build tree
- Added `SDSTORE_HOSTNAME` and  `SDSTORE_NAMESPACE` in config

screen shot 2017-03-08 at 6 42 05 pm

related to https://github.com/screwdriver-cd/screwdriver/issues/468

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "dependencies": {
     "ansi_up": "^1.3.0",
     "humanize-duration": "^3.9.1",
-    "jquery-highlight": "~3.2.2"
+    "jquery-highlight": "~3.2.2",
+    "jstree": "^3.3.3"
```
Comment (minz1027) [inline bower.json]: ember-cli-jstree need this as dependency.

**Comment by tkyi** [comment]
So each build page will have its own artifacts section?
What happens if you click on an artifact?

**Comment by d2lam** [comment]
@tkyi  Yeah, artifacts are per build. And each one is a link, it should take you to the artifact link.

File: app/build-artifact/service.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  let parts;
+
+  if (!tree.children) {
+    parts = tree;
+  } else {
+    parts = tree.children;
+  }
+
+  parts.forEach((part) => {
+    if (part.children.length === 0) {
```
Comment (FenrirUnbound) [inline app/build-artifact/service.js]: What happens with empty directories?

File: app/build-artifact/service.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      part.type = 'file';
+      delete part.children;
+    } else {
+      delete part.a_attr;
```
Comment (FenrirUnbound) [inline app/build-artifact/service.js]: Why is this significant?

Comment (d2lam) [inline app/build-artifact/service.js]: We only what it to be the link to the store if it's a file. If it's a directory then remove the link and call changeFiles recursively.

Comment (FenrirUnbound) [inline app/build-artifact/service.js]: Could that be documented in one of the function docs? It's not clear whether that's an intended feature

File: app/build-artifact/service.js (hunk 2 above)
Comment (minz1027) [inline app/build-artifact/service.js]: Is it a common case? Someone just create a folder under ARTIFACTS and do no write anything in it? If that's the case, there is no way for the UI to tell it's a file or directory. We don't provide any extra info in the manifest, `/emptydir` can be a dir and can be a file. File is the best guess here.

Comment (FenrirUnbound) [inline app/build-artifact/service.js]: It's not rare, but it's an uncommon scenario in V3.

What happens to the link if it's an empty directory?

Comment (minz1027) [inline app/build-artifact/service.js]: You get 404.

Comment (FenrirUnbound) [inline app/build-artifact/service.js]: That behavior should be documented or captured in an issue

**Pull request merged**